crates/executor/src/lib.rs

```diff
@@ -52,25 +58,11 @@ where
     F: TrieDBFetcher,
     H: TrieDBHinter,
 {
-    /// Constructs a new [StatelessL2BlockExecutor] with the given starting state root, parent hash,
-    /// and [TrieDBFetcher].
-    pub fn new(
-        config: &'a RollupConfig,
-        parent_header: Sealed<Header>,
-        fetcher: F,
-        hinter: H,
-    ) -> Self {
-        let trie_db = TrieDB::new(parent_header.state_root, parent_header, fetcher, hinter);
-        let state = StateBuilder::new_with_database(trie_db).with_bundle_update().build();
-        Self { config, state }
+    /// Constructs a new [StatelessL2BlockExecutorBuilder] with the given [RollupConfig].
+    pub fn builder(config: &'a RollupConfig) -> StatelessL2BlockExecutorBuilder<'a, F, H> {
+        StatelessL2BlockExecutorBuilder::with_config(config)
```

Comment:
I like this associated function, avoids needing to reference the Builder directly.